lib/Mojo/IRC.pm

```diff
@@ -594,6 +645,41 @@ CHUNK:
   }
 }
 
+sub _write_and_wait {
```

Comment:
This could also be a public method, but I'm not sure about the API yet, so I'm keeping it private. Questions:
* Should it also take a timeout value in case none of the events match?
* Should the return callback retrieve ```($self, $event, $err, $msg)``` or just ```($self, $event, $msg)```?